### PR TITLE
Add apiUrl to DiscoveredClusterSpec

### PR DESCRIPTION
**Pull request by JakobGray**
Issue: https://github.com/open-cluster-management/backlog/issues/12445

### CODE REVIEW

**Review by nathanweatherly — APPROVED** [review]

File: api/v1alpha1/discoveredcluster_types.go [hunk 1]
```diff
@@ -28,6 +28,7 @@ type DiscoveredClusterSpec struct {
 	Name              string       `json:"name" yaml:"name"`
 	DisplayName       string       `json:"displayName" yaml:"displayName"`
 	Console           string       `json:"console,omitempty" yaml:"console,omitempty"`
+	APIURL            string       `json:"apiUrl,omitempty" yaml:"apiUrl,omitempty"`
```
Comment (zkayyali812) [inline api/v1alpha1/discoveredcluster_types.go:31]: Was this intended to be a required field? If so should we remove omitempty?

Comment (JakobGray) [reply api/v1alpha1/discoveredcluster_types.go]: I've updated it to be required. It may still be an empty string, but it won't be omitted.

**Pull request merged**